src/library/scala/runtime/BoxesRunTime.java

```diff
@@ -115,8 +115,7 @@ public static double unboxToDouble(Object d) {
     /* COMPARISON ... COMPARISON ... COMPARISON ... COMPARISON ... COMPARISON ... COMPARISON */
 
     public static boolean equals(Object x, Object y) {
-        if (x == y) return true;
-        return equals2(x, y);
+        return (x == y) || equals2(x, y);
```

Comment:
The context has excessive "early returns". I don't know whether this is a form of premature java optimization, or just to make java look bad. Here, I would count it as "ambient style" and either fix the entire file (having shown that it doesn't change the bytecode) or just leave it.

For example, `unboxToBoolean` could be rewritten `return b != null && something.booleanValue`. If the rewrite doesn't make it faster, does it make it easier to read?

Reply:
I think that it does make it easier to read.